webapp/src/main/java/life/qbic/views/MainLayout.java

```diff
@@ -17,65 +13,34 @@
  */
 @PageTitle("Data Manager ")
 @Route(value = "data")
-public class MainLayout extends AppLayout {
+public class MainLayout extends DataManagerLayout {
 
-  public Button register;
-  public Button login;
+  public Button logout;
+  private HorizontalLayout loggedInButtonLayout;
 
-  private HorizontalLayout buttonLayout;
-  private HorizontalLayout headerLayout;
-
-  public MainLayout(@Autowired MainHandlerInterface mainHandlerInterface) {
-    createHeaderContent();
-    registerToHandler(mainHandlerInterface);
+  public MainLayout(@Autowired MainHandlerInterface startHandlerInterface) {
+    createNavBarContent();
+    registerToHandler(startHandlerInterface);
   }
 
-
-  private void registerToHandler(MainHandlerInterface mainHandler) {
-    if (mainHandler.handle(this)) {
-      System.out.println("Registered main layout handler");
-    } else {
-      System.out.println("Already registered main layout handler");
-    }
+  private void registerToHandler(MainHandlerInterface startHandler) {
+    startHandler.handle(this);
   }
 
-
-  private void createHeaderContent() {
-    createHeaderLayout();
+  private void createNavBarContent() {
     createHeaderButtonLayout();
 
-    addToNavbar(headerLayout, buttonLayout);
-  }
-
-  private void createHeaderLayout() {
-    H1 appName = styleHeaderTitle();
-    headerLayout = new HorizontalLayout(appName);
-
-    styleHeaderLayout();
-  }
-
-  private void styleHeaderLayout() {
-    headerLayout.setDefaultVerticalComponentAlignment(FlexComponent.Alignment.CENTER);
-    headerLayout.setWidth("100%");
-    headerLayout.addClassNames("py-0", "px-m");
-  }
-
-  private H1 styleHeaderTitle() {
-    H1 appName = new H1("Data Manager");
-    appName.addClassNames("text-l", "m-m");
-    return appName;
+    addToNavbar(loggedInButtonLayout);
```

Comment:
```suggestion
    loggedInButtonLayout = createHeaderButtonLayout()
    addToNavbar(loggedInButtonLayout);
```
Otherwise it is unclear in this scope whether the field already was set.

Reply:
Ignore the red text. :)